test/validator/utils.dart

```diff
@@ -10,8 +10,9 @@ void expectNoValidationError(ValidatorCreator fn) {
   expect(validatePackage(fn), completion(pairOf(isEmpty, isEmpty)));
 }
 
-void expectValidationError(ValidatorCreator fn) {
-  expect(validatePackage(fn), completion(pairOf(isNot(isEmpty), anything)));
+void expectValidationError(ValidatorCreator fn,
+    {Matcher errorsMatcher = isNotEmpty}) {
+  expect(validatePackage(fn), completion(pairOf(errorsMatcher, anything)));
```

Comment:
Rather than doing this, I'd add an `expectSizeValidationError()` function to `size_test.dart`. See `dependency_test.dart`'s `expectDependencyValidationWarning()`.

Reply:
Is it too general? Adding `expectSizeValidationError` (similar to `expectDependencyValidationWarning`) will result in almost same code.